packages/it/index.js

```diff
@@ -1,97 +1,99 @@
-    assign({}, statics, staticProperties) : undefined;
-
-  var sdp = isObject(deepStatics) || isObject(staticDeepProperties) ? merge({}, deepStatics, staticDeepProperties) : undefined;
-
-  var spd = descr.staticPropertyDescriptors;
+  const {
+    methods,
+    properties,
+    props,
+    initializers,
+    init,
+    composers,
+    deepProperties,
+    deepProps,
+    propertyDescriptors,
+    staticProperties,
+    statics,
+    staticDeepProperties,
+    deepStatics,
+    configuration,
+    conf,
+    deepConfiguration,
+    deepConf,
+  } = descr;
+
+  const p = isObject(props) || isObject(properties) ? assign({}, props, properties) : undefined;
+
+  const dp = isObject(deepProps) || isObject(deepProperties) ? merge({}, deepProps, deepProperties) : undefined;
+
+  const sp = isObject(statics) || isObject(staticProperties) ? assign({}, statics, staticProperties) : undefined;
+
+  const sdp =
+    isObject(deepStatics) || isObject(staticDeepProperties) ? merge({}, deepStatics, staticDeepProperties) : undefined;
+
+  let spd = descr.staticPropertyDescriptors;
   if (isString(descr.name)) spd = assign({}, spd || {}, { name: { value: descr.name } });
 
-  var c = isObject(conf) || isObject(configuration) ?
-    assign({}, conf, configuration) : undefined;
+  const c = isObject(conf) || isObject(configuration) ? assign({}, conf, configuration) : undefined;
 
-  var dc = isObject(deepConf) || isObject(deepConfiguration) ? merge({}, deepConf, deepConfiguration) : undefined;
+  const dc = isObject(deepConf) || isObject(deepConfiguration) ? merge({}, deepConf, deepConfiguration) : undefined;
 
-  var ii = extractFunctions(init, initializers);
+  const ii = extractFunctions(init, initializers);
 
-  var cc = extractFunctions(composers);
+  const cc = extractFunctions(composers);
 
-  var descriptor = {};
+  const descriptor = {};
   if (methods) descriptor.methods = methods;
   if (p) descriptor.properties = p;
   if (ii) descriptor.initializers = ii;
   if (cc) descriptor.composers = cc;
   if (dp) descriptor.deepProperties = dp;
   if (sp) descriptor.staticProperties = sp;
   if (sdp) descriptor.staticDeepProperties = sdp;
-  if (pd) descriptor.propertyDescriptors = pd;
+  if (propertyDescriptors) descriptor.propertyDescriptors = propertyDescriptors;
   if (spd) descriptor.staticPropertyDescriptors = spd;
   if (c) descriptor.configuration = c;
   if (dc) descriptor.deepConfiguration = dc;
 
   return descriptor;
-}
+};
 
-function stampit() {
-  'use strict'; // to make sure `this` is not pointing to `global` or `window`
-  var length = arguments.length, args = [];
-  for (var i = 0; i < length; i += 1) {
-    var arg = arguments[i];
-    args.push(isStamp(arg) ? arg : standardiseDescriptor(arg));
-  }
-
-  return compose.apply(this || baseStampit, args); // jshint ignore:line
+function stampit(...args) {
+  return compose.apply(this || baseStampit, args.map((arg) => (isStamp(arg) ? arg : standardiseDescriptor(arg))));
```

Comment:
This was also a perf optimisation. :) But this function is not the hot area. So converting to `map()` in this case has benefits like shorter code. 👍